app/modules/parsing/graph_construction/parsing_helper.py

```
@@ -353,23 +430,39 @@ async def setup_project_directory(
+                    extracted_dir = await self.download_and_extract_tarball(
+                        repo,
+                        branch,
+                        os.getenv("PROJECT_PATH"),
+                        auth,
+                        repo_details,
+                        user_id,
+                    )
+                    branch_details = repo_details.get_branch(branch)
+                    latest_commit_sha = branch_details.commit.sha
+            except ParsingFailedError as e:
+                logger.error(f"Failed to download repository: {e}")
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)